main/lsp/requests/code_action.cc

```diff
@@ -192,98 +192,104 @@ unique_ptr<ResponseMessage> CodeActionTask::runRequest(LSPTypecheckerDelegate &t
-            auto arg0Loc = core::Loc(file, resp->argLocOffsets[0]);
-            auto newContents = arg0Loc.source(gs).value();
+            } else if (auto *resp = isTUnsafeOrMustResponse(gs, queryResult.responses)) {
+                auto tdi =
+                    make_unique<VersionedTextDocumentIdentifier>(move(params->textDocument->uri), JSONNullObject());
+                auto replaceRange = Range::fromLoc(gs, resp->termLoc());
+                auto arg0Loc = core::Loc(file, resp->argLocOffsets[0]);
+                auto newContents = arg0Loc.source(gs).value();
 
-            vector<unique_ptr<TextEdit>> edits;
-            edits.emplace_back(make_unique<TextEdit>(move(replaceRange), string(newContents)));
+                vector<unique_ptr<TextEdit>> edits;
+                edits.emplace_back(make_unique<TextEdit>(move(replaceRange), string(newContents)));
 
-            vector<unique_ptr<TextDocumentEdit>> documentEdits;
-            documentEdits.emplace_back(make_unique<TextDocumentEdit>(move(tdi), move(edits)));
+                vector<unique_ptr<TextDocumentEdit>> documentEdits;
+                documentEdits.emplace_back(make_unique<TextDocumentEdit>(move(tdi), move(edits)));
 
-            auto workspaceEdit = make_unique<WorkspaceEdit>();
-            workspaceEdit->documentChanges = move(documentEdits);
+                auto workspaceEdit = make_unique<WorkspaceEdit>();
+                workspaceEdit->documentChanges = move(documentEdits);
 
-            auto action = make_unique<CodeAction>(fmt::format("Delete T.{}", resp->callerSideName.show(gs)));
-            action->kind = CodeActionKind::RefactorRewrite;
-            action->edit = move(workspaceEdit);
-            result.emplace_back(move(action));
-        } else {
-            if (loc.beginPos() != loc.endPos() && config.opts.lspExtractToVariableEnabled) {
-                // For move method to new module we use canResolveLazily to defer the computation
-                // until the user has actually selected the action. We can't do that here because
-                // we need to do the core computation to know if extract the current selection is
-                // valid in the first place, to decide if we can show the code action or not.
-                Timer timeit(gs.tracer(), "Extract to Variable");
-
-                auto documentEdits = VariableExtractor::getEdits(typechecker, config, loc);
-                if (!documentEdits.empty()) {
-                    auto action = make_unique<CodeAction>("Extract Variable");
-                    action->kind = CodeActionKind::RefactorExtract;
+                auto action = make_unique<CodeAction>(fmt::format("Delete T.{}", resp->callerSideName.show(gs)));
+                action->kind = CodeActionKind::RefactorRewrite;
+                action->edit = move(workspaceEdit);
+                result.emplace_back(move(action));
+            } else {
+            }
```

Comment:
Looks like you've got an empty `else` block, is that expected?
```cpp
} else {
}
```

Reply:
Good catch, I forgot to delete that after moving the code in it. I've deleted it.